### ISSUE TITLE
Update Notification transition crashes sentinel if the patient id is misconfigured

### ISSUE THREAD

**Issue by SCdF** [medic/medic-webapp#4121]
For project-specific information, see: https://github.com/medic/medic-projects/issues/3240

```
medic_medic_medic-sentinel54.log:Error: Cannot provide registrations to template context without a patient
medic_medic_medic-sentinel54.log-    at Error (native)
medic_medic_medic-sentinel54.log-    at extendedTemplateContext (/srv/storage/gardener/data/working_dir/aHR0cDovL2xvY2FsaG9zdDo1OTg0L21lZGljL19kZXNpZ24vbWVkaWMvbWVkaWMtc2VudGluZWw=/node_modules/medic-sentinel/lib/messages.js:30:15)
medic_medic_medic-sentinel54.log-    at Object.addMessage (/srv/storage/gardener/data/working_dir/aHR0cDovL2xvY2FsaG9zdDo1OTg0L21lZGljL19kZXNpZ24vbWVkaWMvbWVkaWMtc2VudGluZWw=/node_modules/medic-sentinel/lib/messages.js:51:27)
medic_medic_medic-sentinel54.log-    at Object._addMsg (/srv/storage/gardener/data/working_dir/aHR0cDovL2xvY2FsaG9zdDo1OTg0L21lZGljL19kZXNpZ24vbWVkaWMvbWVkaWMtc2VudGluZWw=/node_modules/medic-sentinel/transitions/update_notifications.js:59:22)
medic_medic_medic-sentinel54.log-    at async.parallel (/srv/storage/gardener/data/working_dir/aHR0cDovL2xvY2FsaG9zdDo1OTg0L21lZGljL19kZXNpZ24vbWVkaWMvbWVkaWMtc2VudGluZWw=/node_modules/medic-sentinel/transitions/update_notifications.js:146:34)
medic_medic_medic-sentinel54.log-    at /srv/storage/gardener/data/working_dir/aHR0cDovL2xvY2FsaG9zdDo1OTg0L21lZGljL19kZXNpZ24vbWVkaWMvbWVkaWMtc2VudGluZWw=/node_modules/medic-sentinel/node_modules/async/dist/async.js:3838:9
medic_medic_medic-sentinel54.log-    at /srv/storage/gardener/data/working_dir/aHR0cDovL2xvY2FsaG9zdDo1OTg0L21lZGljL19kZXNpZ24vbWVkaWMvbWVkaWMtc2VudGluZWw=/node_modules/medic-sentinel/node_modules/async/dist/async.js:421:16
medic_medic_medic-sentinel54.log-    at iterateeCallback (/srv/storage/gardener/data/working_dir/aHR0cDovL2xvY2FsaG9zdDo1OTg0L21lZGljL19kZXNpZ24vbWVkaWMvbWVkaWMtc2VudGluZWw=/node_modules/medic-sentinel/node_modules/async/dist/async.js:928:24)
medic_medic_medic-sentinel54.log-    at /srv/storage/gardener/data/working_dir/aHR0cDovL2xvY2FsaG9zdDo1OTg0L21lZGljL19kZXNpZ24vbWVkaWMvbWVkaWMtc2VudGluZWw=/node_modules/medic-sentinel/node_modules/async/dist/async.js:906:16
medic_medic_medic-sentinel54.log-    at /srv/storage/gardener/data/working_dir/aHR0cDovL2xvY2FsaG9zdDo1OTg0L21lZGljL19kZXNpZ24vbWVkaWMvbWVkaWMtc2VudGluZWw=/node_modules/medic-sentinel/node_modules/async/dist/async.js:3835:13
```

This occurs if the patient is not attached correctly.

It's not clear if the solution is to change the Error into a validation error that attaches to the document, or if the code should not get this far with a misconfigured patient.

**Comment by SCdF** [comment]
@mukesh2006 if you're interested in performing open heart surgery on a server which is having issues may I suggest the following incision:

```diff
diff --git a/transitions/update_notifications.js b/transitions/update_notifications.js
index 2c4add5..70b80a1 100644
--- a/transitions/update_notifications.js
+++ b/transitions/update_notifications.js
@@ -136,7 +136,7 @@ module.exports = {
                     return callback(err);
                 }
 
-                if (registrations.length) {
+                if (registrations.length && patient) {
                     if (eventType.mute) {
                         if (config.confirm_deactivation) {
                             self._addErr('confirm_deactivation', config, doc);
```

NB: I have yet to test things locally, but this could be it. It could also very much not be it. Warranty not included, etc. Talk to me when you're online.

**Comment by SCdF** [comment]
Two PRs, one against master, one against 2.13.x. Will CP the 2.13.x into 2.14.x as well once it's approved.

**Comment by SCdF** [comment]
@alxndrsn can you review the two linked PRs please?

**Comment by SCdF** [comment]
Merged into master, 2.14.x and 2.13.x. Soon to be released in a 2.13.x release

**Comment by SCdF** [comment]
This has been released in 2.13.5
